lib/utils/cli.go

```diff
@@ -283,6 +284,11 @@ func Consolef(w io.Writer, log logrus.FieldLogger, component, msg string, params
 // some defaults common for all Teleport CLI tools
 func InitCLIParser(appName, appHelp string) (app *kingpin.Application) {
 	app = kingpin.New(appName, appHelp)
+	// prevent Kingpin from calling os.Exit(), we want to handle errors ourselves.
+	app.Terminate(nil)
```

Comment:
I don't see any  the `ox.Exit` call added so in case of invalid flags or arguments like `tsh foo` the tsh will return `0` as exit status code ?

Reply:
No, the error code will be non-zero still. The thing is, without `app.Terminate(nil)` Kingpin will, in some cases, terminate the entire app instead of simply returning an error.

Reply:
Fixed in: dd26cf68919ebd92df3833a292958e597f79a85e